warc2zim/main.py

```diff
@@ -0,0 +1,309 @@
+            raise
+
+    def get_url(self):
+        return 'A/' + self.filename
+
+    def get_mime_type(self):
+        return self.mime
+
+    def get_data(self):
+        return Blob(self.content)
+
+
+# ============================================================================
+class RWPViewerArticle(BaseArticle):
+    def __init__(self, filename, main_url):
+        super(RWPViewerArticle, self).__init__()
+        self.filename = filename
+        self.main_url = main_url
+
+    def get_url(self):
+        return 'A/' + self.filename
+
+    def get_mime_type(self):
+        return 'text/html'
+
+    def get_data(self):
+        content = """
+<html>
+  <head>
+    <style>
+    body {{
+      width: 100%;
+      height: 100%;
+      overflow-y: hidden;
+      margin: 0px;
+      padding: 0px;
+    }}
+    </style>
+    <script src="./ui.js"></script>
+  </head>
+  <body>
+    <replay-web-page
+     source="proxy:../"
+     config='{{"type": "kiwix"}}'
+     replayBase="./"
+     url="{0}"
+     embed="replayonly"
+     deepLink="true"
+     />
+  </body>
+</html>
+""".format(self.main_url)
+
+        return Blob(content.encode('utf-8'))
+
+
+# ============================================================================
+class WARC2Zim:
+    REPLAY_STATIC_FILES = ['index.html', 'ui.js', 'sw.js']
+
+    def __init__(self, args):
+        logging.basicConfig(format='[%(levelname)s] %(message)s')
+        if args.verbose:
+            logger.setLevel(logging.DEBUG)
+        else:
+            logger.setLevel(logging.INFO)
+
+        self.indexed_urls = set({})
+        self.name = args.name
+        if not self.name:
+            self.name, ext = os.path.splitext(args.inputs[0])
+            self.name += '.zim'
+
+        self.inputs = args.inputs
+        self.replay_viewer_source = args.replay_viewer_source
+        self.main_url = args.main_url
+
+        self.replay_articles = []
+        self.revisits = {}
+
+    def run(self):
+        for filename in self.REPLAY_STATIC_FILES:
+            self.replay_articles.append(RWPStaticArticle(self.replay_viewer_source, filename))
+
+        with Creator(self.name, main_page='viewer.html', index_language='', min_chunk_size=8192) as zimcreator:
+            # add replay system
+            for article in self.replay_articles:
+                zimcreator.add_article(article)
+
+            zimcreator.add_article(RWPViewerArticle('viewer.html', self.main_url))
+
+            for warcfile in self.inputs:
+                self.warc2zim(warcfile, zimcreator)
+
+            # process revisits, headers only
+            for url, record in self.revisits.items():
+                if url not in self.indexed_urls:
+                    logger.debug('Adding revisit {0} -> {1}'.format(url, record.rec_headers['WARC-Refers-To-Target-URI']))
+                    zimcreator.add_article(WARCHeadersArticle(record))
+
+    def warc2zim(self, warcfile, zimcreator):
```

Comment:
Not fan of the method name. Think we're loosing meaning by multiplying similar-name methods.

Reply:
Sure, will change

Reply:
Renamed to process_warc()